src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -1634,6 +1656,27 @@ void TrackerOnCollectionChanged(object sender, EventArgs eventArgs)
 				UpdateToolbarItems();
 			}
 
+			void OnToolbarItemPropertyChanged(object sender, PropertyChangedEventArgs e)
+			{
+				// Only rebuild toolbar items if a relevant property changed
+				if (e.PropertyName == MenuItem.IsEnabledProperty.PropertyName ||
+					e.PropertyName == MenuItem.TextProperty.PropertyName ||
+					e.PropertyName == MenuItem.IconImageSourceProperty.PropertyName)
+				{
+					// Throttle updates to prevent excessive rebuilding when multiple properties change rapidly
+					if (!_toolbarUpdatePending)
+					{
+						_toolbarUpdatePending = true;
```

Comment:
Do we really need this? What happens if removed?

Reply:
Unfortunately we do need this. The menu in iOS is immutable once you created it. So if we want to change any of these things we need to rebuild the menu entirely.
Removing and adding works, updated the sample to show this.